### PR TITLE
Add medium padding token and document size ramp

### LINKED ISSUE #1518: Create token for 8px
## 😯 Problem to Solve

In a few different places throughout our code and client code, we want a spacing of 8px. Therefore, we often do things like `padding: calc($ni-nimble-standard-padding / 2);` or `padding: calc($ni-nimble-small-padding * 2);`. If 8px is going to be a common value in our designs (which it seems like it is already), then we should have a first-class way of using that value without having to calculate it from other token values.

## 💁 Proposed Solution

Create a new token with a value of `8px`. I'm not sure what we want to name it, but it could be something like `$ni-nimble-medium-padding` or `$ni-nimble-half-standard-padding`.

### PR DESCRIPTION
**Pull request by m-akinc**
# Pull Request

## 🤨 Rationale

Fixes #1518

## 👩‍💻 Implementation

- defined new `mediumPadding` token (8px)
- replaced usages of `standardPadding / 2` and `smallPadding * 2` in our stylesheets
- changed token descriptions of `smallPadding` and `standardPadding` to better reflect that they are fixed-size
- created new Size Ramp story under the Tokens section

## 🧪 Testing

None

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Comment by m-akinc** [comment]
@atmgrifter00 could I please get a buddy on this?

File: packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts
Comment (m-akinc) [inline packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts:1]: I did not remove the three size ramp tokens from the existing Theme-Aware Tokens story. I wasn't sure if we wanted it to still be the full set of tokens or if it should exclude those that are in the Size Ramp story.

**Review by atmgrifter00 — APPROVED** [review]

**Review by jattasNI — APPROVED** [review]

File: packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+            td { 
+                padding: 10px;
+                height: 32px;
+            }
+        </style>
+        <table>
+            <thead>
+                <tr>
+                    <th>${x => x.propertyFormat} Property</th>
+                    <th>Value</th>
```
[the remark this answers is not in the document]
Comment (m-akinc) [reply packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts:90]: IMO we shouldn't have that kind of guidance for these fixed size ramp tokens. We should ultimately have context-specific tokens for those use cases. These tokens, by definition, have no designated applications. If anything, we should provide general guidance at the top of the page that applies to all the size ramp tokens, like "Use these tokens for alignment and spacing when no designated token exists for the purpose." Disagree?

File: packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                font: var(${groupHeaderFont.cssCustomProperty});
+                color: var(${groupHeaderFontColor.cssCustomProperty});
+                text-transform: var(${groupHeaderTextTransform.cssCustomProperty});
+            }
+            td { 
+                padding: 10px;
+                height: 32px;
+            }
+        </style>
+        <table>
```
[the remark this answers is not in the document]
Comment (m-akinc) [reply packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts:86]: I went ahead and updated my new story to use nimble-table, but then when I went to update the Theme-Aware Tokens story, I realized we don't have a column to support this:
image
So..... I guess we should keep consistent and leave them both using the HTML table. 🫤

File: packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jattasNI) [reply packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts:90]: Fair point, I like your suggestion to have general guidance.

File: packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts
Comment (m-akinc) [reply packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts:1]: I did it, but it's a little weird. See if you like it. I now link any description that includes "size ramp token" to the Size Ramp Tokens story.

**Review by rajsite — APPROVED** [review]

File: packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-akinc) [reply packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts:90]: Added.

File: packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts
Comment (jattasNI) [reply packages/nimble-components/src/theme-provider/tests/size-ramp.stories.ts:1]: Hmm, yeah, I agree the implementation seems a little fragile. If we ever change the descriptions it seems likely those links would just break without us noticing. I'd vote to remove the links.

Instead we could consider just updating the comment of those four tokens to be something like this, with no link: "Fixed 4px size ramp token for component layout. See Size Ramp docs for more info"

But I wouldn't complain if you leave the descriptions as they are now.

**Review by jattasNI — APPROVED** [review]

**Pull request merged**